tools/local-sensor/main.go

```diff
@@ -419,15 +421,18 @@ func setupCentralWithRealConnection(cli client.Interface, localConfig localSenso
 	utils.CrashOnError(os.Setenv("ROX_CENTRAL_ENDPOINT", localConfig.CentralEndpoint))
 
 	clientconn.SetUserAgent(clientconn.Sensor)
-	centralConnFactory, err := centralclient.NewCentralConnectionFactory(env.CentralEndpoint.Setting())
+
+	centralClient, err := centralclient.NewClient(env.CentralEndpoint.Setting())
 	if err != nil {
-		utils.CrashOnError(errors.Wrapf(err, "sensor failed to start while initializing gRPC client to endpoint %s", env.CentralEndpoint.Setting()))
+		utils.CrashOnError(errors.Wrapf(err, "sensor failed to start while initializing HTTP client to endpoint %s", env.CentralEndpoint.Setting()))
 	}
+	centralConnFactory := centralclient.NewCentralConnectionFactory(centralClient)
```

Comment:
Is there a reason why the CertLoader is not passed to the connection factory? 
I think the cert loader could be passed directly to the connection factory, this would save passing around the variable some times.

Reply:
Yes, there's a special case for the initial connection where I tried to reuse certificates between gRPC and HTTP clients and make just a single TLS challenge remote call. That's why I passed CertLoader as parameter.